src/backend_model.cc

```diff
@@ -1356,36 +1356,40 @@ TRITONBACKEND_StateBuffer(
   SequenceState* to = reinterpret_cast<SequenceState*>(state);
   Status status = Status::Success;
 
-  // If the buffer size exactly matches the buffer available, reuse the
-  // currently allocated buffer.
-  if (to->Data()->TotalByteSize() == buffer_byte_size) {
-    const std::shared_ptr<AllocatedMemory>& memory =
-        reinterpret_cast<const std::shared_ptr<AllocatedMemory>&>(to->Data());
-
-    TRITONSERVER_MemoryType current_memory_type;
-    int64_t current_memory_type_id;
-    void* lbuffer =
-        memory->MutableBuffer(&current_memory_type, &current_memory_type_id);
-
-    // If the requested memory type doesn't match the current buffer, allocate
-    // a new buffer with the requested memory type and memory type id.
-    if (current_memory_type == *memory_type &&
-        current_memory_type_id == *memory_type_id) {
-      *buffer = lbuffer;
+  TRITONSERVER_MemoryType current_memory_type;
```

Comment:
Question: Could we refactor this into the SequenceState class?

Provide a "Resize()" method on SequenceState class that would internally handle the different types of state. 

That would also enable us to reduce the public interface of the class and handle them internally.

Reply:
Done.